### PR TITLE
fix random rndints for 64 bits address space

### PR DESCRIPTION
**Pull request by luzhixing12345**
int rand() could only generate random number in 0-2^31-1, but the region size is size_t(64B), not all addresses in the range could get accessed.

### CODE REVIEW

**Comment by sjp38** [comment]
Could you please make the description of the change bit more detailed?  What is the problem, and how this change is fixing it?  Please add at least the PR description to the commit message.

File: masim.c [hunk 1]
```diff
@@ -56,7 +56,7 @@ void pr_phase(struct phase *phase)
 	struct access *pattern;
 	int j;
 
-	printf("Phase (%s) for %u ms\n", phase->name, phase->time_ms);
+	printf("Phase (%s) for %lu ms\n", phase->name, phase->time_ms);
```
Comment (sjp38) [inline masim.c]: This change is for addresses, not time, right?  I think this change is not really necessary?

File: masim.c [hunk 2]
```diff
@@ -130,7 +134,7 @@ static void do_seq_ro(struct access *access, int batch)
 {
 	struct mregion *region = access->mregion;
 	char *rr = region->region;
-	size_t offset = access->last_offset;
+	uint64_t offset = access->last_offset;
```
Comment (sjp38) [inline masim.c]: `size_t` is 8 bytes on 64-bit machines, right?  Would we want to run 64-bit access pattern on 32-bit machine?  If not, why we need this change?

File: masim.c [hunk 3]
```diff
@@ -492,7 +496,7 @@ int parse_phase(char *lines[], int nr_lines, struct phase *p,
 
 	p->name = (char *)malloc((strlen(lines[0]) + 1) * sizeof(char));
 	strcpy(p->name, lines[0]);
-	p->time_ms = atoi(lines[1]);
+	p->time_ms = atoll(lines[1]);
```
Comment (sjp38) [inline masim.c]: The commit message was saying this change is for 64bit address, not time.  But I think this change is for supporting 64-bit time values?  Why we need this change?

File: masim.c [hunk 4]
```diff
@@ -509,13 +513,14 @@ int parse_phase(char *lines[], int nr_lines, struct phase *p,
 		for (k = 0; k < nr_regions; k++) {
 			if (strcmp(fields[0], regions[k].name) == 0) {
 				a->mregion = &regions[k];
+                break;
```
Comment (sjp38) [inline masim.c]: Why `break` here?

File: masim.h [hunk 5]
```diff
@@ -1,9 +1,11 @@
 #ifndef _MASIM_H
 #define _MASIM_H
 
+#include <stdint.h>
+
 struct mregion {
 	char name[256];
-	size_t sz;
+	uint64_t sz;
```
Comment (sjp38) [inline masim.h]: Again, isn't size of `size_t` on 64 bit machines are 8 bytes?  Why we need this change?

File: masim.h [hunk 6]
```diff
@@ -16,18 +18,18 @@ enum rw_mode {
...
 	/* For runtime only */
 	int prob_start;
-	size_t last_offset;
+	uint64_t last_offset;
 };
 
 struct phase {
 	char *name;
-	unsigned time_ms;
+	uint64_t time_ms;
```
Comment (sjp38) [inline masim.h]: Questions about `size_t` size of 64bit machines and 64bit support of time can be applied to above three changes, too.

File: masim.c [hunk 7]
```diff
@@ -189,7 +193,7 @@ static void do_seq_rw(struct access *access, int batch)
 {
 	struct mregion *region = access->mregion;
 	char *rr = region->region;
-	size_t offset = access->last_offset;
+	uint64_t offset = access->last_offset;
```
Comment (sjp38) [inline masim.c]: Questions about `size_t` size of 64bit machines apply to above three changes, too.

File: masim.c (hunk 4 above)
Comment (luzhixing12345) [reply masim.c]: A pattern corresponds to only one area of memory. So if match the corresponding region name, there's no need to check for others.

**Comment by luzhixing12345** [comment]
Sorry for my mistake in thinking that size_t was a signed 8-byte type. There's no need to change size_t to uint64_t.
unsigned time_ms is also enough.

Random number type should be change to size_t, this PR fixed it by generating a 64-bit random number (combining 2 32-bit random numbers basically), because you can see that all addresses in the range may not get accessed

https://github.com/sjp38/masim/blob/7f2c42535e31c75855f47cf4c56df6e63cca5ee2/masim.c#L90-L116

File: masim.c (hunk 4 above)
Comment (sjp38) [reply masim.c]: Thanks for the clarification.  But this is a change that is not related with the 64bit fix, right?  Could you please separate this fix to another commit, and adjust the PR message?

Comment (luzhixing12345) [reply masim.c]: ok, sure

**Comment by sjp38** [comment]
Thank you posting and revising this PR!  Merged.

**Pull request merged**